ash/src/util.rs

```diff
@@ -59,6 +60,15 @@ impl<T> Align<T> {
         }
     }
 
+    pub unsafe fn index(&self, index: Range<usize>) -> Align<T> {
+        Align {
+            ptr: (self.ptr as *mut u8).offset((index.start as u64 * self.elem_size) as isize) as *mut c_void,
+            elem_size: self.elem_size,
+            size: (index.end - index.start) as u64 * self.elem_size as u64,
+            _m: self._m,
```

Comment:
You should be able to use struct update syntax to carry `elem_size` and `_m` over:

```suggestion
            ptr: (self.ptr as *mut u8).offset((index.start as u64 * self.elem_size) as isize) as *mut c_void,
            size: (index.end - index.start) as u64 * self.elem_size as u64,
            ..*self
```